battle-engine.js

```diff
@@ -325,22 +317,22 @@ BattlePokemon = (() => {
 			stat = this.battle.getStatCallback(stat, statName, this, unboosted);
 		}
 		return stat;
-	};
-	BattlePokemon.prototype.getDecisionSpeed = function () {
+	}
+	get decisionSpeed() {
```

Comment:
That is something I actually do want you to change back. Don't use getters for things that aren't supposedly immutable.